Stokesian Dynamics (#3790)
espressomd/espresso: doc/sphinx/running.rst

for backward compatibility. New scripts should setup the steepest descent
integrator with the :meth:`~espressomd.integrate.IntegratorHandle.set_steepest_descent`
handle directly.

.. _Stokesian Dynamics:

Stokesian Dynamics
------------------

:meth:`espressomd.integrate.IntegratorHandle.set_stokesian_dynamics`

The Stokesian Dynamics method allows to study the behaviour of spherical
particles in a viscous fluid. It is targeted at systems with very low Reynolds

> numbers. In such systems, particles stop moving immediately as soon as any

Reviewer: Not a property of the system but an assumption/approximation?

Reviewer 2: In the limit of zero Reynolds number the statement is exact.  Maybe a better formulation would be
```suggestion
numbers. In such systems, particles come to a rest almost immediately as soon as any
```

force on them is removed. In other words, motion has no memory of the past.

> The integration scheme is relatively simple. Only the particle's positions,

Reviewer: positions'?

Reviewer 2: Did you mean `particles'`?
```suggestion
The integration scheme is relatively simple. Only the particles' positions,
```

:py:meth:`espressomd.integrate.IntegratorHandle.set_stokesian_dynamics()`.

Note that this setup represents a system at zero temperature. In order to
thermalize the system, the SD thermostat needs to be activated (see
:ref:`Stokesian thermostat`).

.. _Important_SD:

Important
~~~~~~~~~

The particles must be prevented from overlapping. It is mathematically allowed
for the particles to overlap to a certain degree. However, once the distance
of the sphere centers is less than 2/3 of the sphere diameter, the mobility
matrix is no longer positive definite and the Stokesian Dynamics integrator
will fail. Therefore, the particle centers must be kept apart from each
other by a strongly repulsive potential, for example the WCA potential
that is set to the appropriate particle radius (for more information about
the available interaction types see :ref:`Non-bonded interactions`).

The current implementation of SD only includes the far field approximation.
The near field (so-called lubrication) correction is planned. For now,
Stokesian Dynamics provides a good approximation of the hydrodynamics
in dilute systems where the average distance between particles is several

> sphere diameters.

Reviewer: Should it be stated that SD is provided via an external library.

Author: I've just added that external dependency info in both the SD integrator and SD thermostat documentation.
